x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -18,6 +17,21 @@ import (
 	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
 )
 
+// ClaimProcessingContext holds the processing state and metadata for a claim during settlement.
+// It tracks whether the claim was successfully settled or expired.
+type ClaimProcessingContext struct {
+	// SettlementResult contains the token logic module operations (mint, burn, transfer)
+	// to be executed for this claim settlement.
+	SettlementResult *tokenomicstypes.ClaimSettlementResult
+	// IsSettled is a flag indicating whether the claim was successfully settled (true)
+	// or expired without settlement (false)
+	IsSettled bool
+	// NumClaimRelays is the total number of relays in the claim's session tree
+	NumClaimRelays uint64
+	// NumClaimComputeUnits is the total compute units claimed by the supplier
+	NumClaimComputeUnits uint64
+}
+
```

Comment:
I added a `numDiscardedFaultyClaims`.

Can you return it and add it to the telemetry?